### ISSUE TITLE
Share sheet?

### ISSUE THREAD

**Issue by ifuchs** [ColdGrub1384/Pyto#139]
Is it possible to invoke a Pyto script from the share sheet? I would like to be able to share a photo or other document with Pyto as input to a script. If this is possible, is there an example of its use available?

**Comment by cvpe** [comment]
I don't think it is already possible, but temporarily, you could pass through Shortcuts app.
Try this [shortcut](https://www.icloud.com/shortcuts/2679dc1aa3104c5e939fec7ede909d79)

**Comment by cvpe** [comment]
But, only Shortcuts app runs in share sheet mode, Pyto is launched in full mode.

**Comment by ifuchs** [comment]
Pythonista does support the share sheet. You can select run Pytonista script and it takes from the share sheet and gives it to Pythonista to script.

**Comment by cvpe** [comment]
Yes, I know that but you asked for Pyto, aren't you?

**Comment by ifuchs** [comment]
Yes of course. I was hoping to get the same capability using Pyto. I need access to pandas and pythonista does not include it

**Comment by cvpe** [comment]
I understand, that's why pass through Shortcuts could be a workaround because it launches Pyto...

**Comment by ifuchs** [comment]
OK, thanks. I just thought perhaps there was a way to add it to Pyto.

**Comment by cvpe** [comment]
I suppose that Pyto author will add it in the future.
But, if it is only to pass a photo, you can pick it in Pyto, but I guess you know it.

**Comment by ifuchs** [comment]
The Shortcut gives the error:

PermissionError

[Errno 1] Operation not permitted: '/private/var/mobile/Library/Mobile Documents/iCloud~is~workflow~my~workflows/Documents/temp_photo.'

Do you know where that is set?

**Comment by cvpe** [comment]
I think you have to first, in Pyto settings, give access to external folder of Shortcuts

**Comment by cvpe** [comment]
![](https://i.imgur.com/6aCdbdp.jpg)

**Comment by ifuchs** [comment]
What I am trying to do is to send a text string, which is actually an SQL query, from another app to Pyto for execution.

**Comment by cvpe** [comment]
How do you plan to send the string?
Via the share sheet? 
Via an url scheme?
Via the clipboard?
And how do you want to launch Pyto?

**Comment by ifuchs** [comment]
I would like to be able to use the share sheet and perhaps at times the
clipboard. As to how Paetow will be started, I would assume the same way
other programs that you can share with are started such as pythonista. That
is, they should not require me to do anything to start the app that is
receiving the share sheet.

**Comment by cvpe** [comment]
I think that nothing has changed, thus Pyto can not yet be started via the share sheet. As already said, you could start another app in the share sheet (Pythonista or Shortcuts) and this app could launch Pyto via its url scheme, passing the string received via the share sheet. The way is longer but if it is the only one...

**Comment by ifuchs** [comment]
I guess what I don’t quite understand is why I can go to the files app and click on a pyto file and it will open in pyto(which was not open before) yet I can’t do the equivalent from a share sheet?

**Comment by cvpe** [comment]
Perhaps if the click file is in the Pyto folder...
